### PR TITLE
Fix EWS-1294 date missing in the CSV download; COUNTRY=cambodia

### LINKED ISSUE #1049: [Bug]: Cambodia - EWS-1294 date field is missing info in the CSV download
### What happened?

When I download a CSV file for EWS-1294 data, the date column shows time of day, but not the full date. We should use the full date + time in this column, or have a separate column for the date.

To replicate, use Cambodia and select Flood > Early warning > EWS-1294. Then click on a data point to see the chart, then click the download button.

### Which country / deployment are you running?

Cambodia on main branch.

### Add a screenshot (if relevant)

Screenshot 2024-01-08 at 16 38 41

### PR DESCRIPTION
**Pull request by echaidemenos**
### Description

This fixes #1049 .

Fixes a bug where only the time would appear in the CSV for Cambodia - EWS-1294 layer

## How to test the feature:

- select Cambodia -> Flood -> Early Warning
- select array to display chart on the map
- use download CSV button and verify the date show both date and time

## Checklist - did you ...

Test your changes with

- [ ] `REACT_APP_COUNTRY=rbd yarn start`
- [ ] `REACT_APP_COUNTRY=cambodia yarn start`
- [ ] `REACT_APP_COUNTRY=mozambique yarn start`
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: frontend/src/components/Common/Chart/index.tsx [hunk 1]
```diff
@@ -27,15 +27,19 @@ interface ChartData {
   }[];
 }
 
-function downloadCsv(chartData: ChartData, filename: string) {
+function downloadCsv(
```
Comment (ericboucher) [inline frontend/src/components/Common/Chart/index.tsx:30]: This is a bit counter-intuitive to me. It doesn't feel like this should be the responsibility of "downloadCSV". Whoever is calling downloadCSV should prepare the data.

Also, it looks like we have two functions `downloadCsv` that do basically the same thing, so we need to homogenize this.

I would actually suggest inverting the behavior, keep the full date in the charts data, but have an option to display time only when we visualize the chart. What do you think?

File: frontend/src/components/Common/Chart/index.tsx [hunk 2]
```diff
@@ -153,16 +157,27 @@ const Chart = memo(
...
+          return moment(row[config.category])
+            .locale(t('date_locale') as LocaleSpecifier)
+            .format('YYYY-MM-DD HH:mm');
+        });
+
+      return data.EWSConfig
+        ? [fullLabels.map(x => x.split(' ')[1]), fullLabels]
+        : [
+            fullLabels.map(x => x.split(' ')[0]),
+            fullLabels.map(x => x.split(' ')[0]),
```
Comment (laurentS) [inline frontend/src/components/Common/Chart/index.tsx:179]: You're returning the same values twice, is this normal? Should the second part be `fullLabels` instead for consistency with above?

File: frontend/src/components/Common/Chart/index.tsx [hunk 3]
```diff
@@ -153,16 +157,27 @@ const Chart = memo(
...
-        return moment(row[config.category])
-          .locale(t('date_locale') as LocaleSpecifier)
-          .format(dateFormat);
-      });
+      const fullLabels = tableRows
+        .slice(chartRange[0], chartRange[1])
+        .map(row => {
+          return moment(row[config.category])
+            .locale(t('date_locale') as LocaleSpecifier)
+            .format('YYYY-MM-DD HH:mm');
```
Comment (laurentS) [inline frontend/src/components/Common/Chart/index.tsx:172]: Is there a way to get rid of `moment` here?

File: frontend/src/components/Common/Chart/index.tsx [hunk 4]
```diff
@@ -366,9 +354,15 @@ const Chart = memo(
...
                 <IconButton
-                  onClick={() =>
-                    downloadCsv(chartData, title.split(' ').join('_'))
-                  }
+                  onClick={() => {
+                    const chartData = {
+                      labels: labels.map(x =>
+                        shouldDisplayTime ? x : String(x).split(' ')[0],
+                      ),
+                      datasets: datasetsTrimmed,
```
Comment (ericboucher) [inline frontend/src/components/Common/Chart/index.tsx:362]: Not sure I understand why this is needed?

**Review by ericboucher — APPROVED** [review]

**Pull request merged**